### PR TITLE
Question Anwering Query

### LINKED ISSUE #252: Question and Answering
Add a Q&A query for NLP. Try to stick to Keras + TF for consistency and use a finetune approach (HuggingFace could be a great place to go)

### PR DESCRIPTION
**Pull request by pranavnt**
---
name: b) New feature - Question Answering Query
about: Select this if you're adding a new feature to libra.
A query that fits a pre-trained BERT QA model to the dataset (more explained later).
labels: feature
---

This pull request closes #252 

**- What I did**
I developed a query that fine-tunes a question answering model on the user's dataset. 

**- How I did it**
I used the cdqa library which allowed me to use and fine tune a pre-trained BERT based QA model. Their repo can be found here: https://github.com/cdqa-suite/cdQA 

**- How to verify it**
Go through the instructions in this colab notebook to run the code: https://colab.research.google.com/gist/pranavnt/96b426f9a44f0c0ef6abf38bf81fbded/qa-testing.ipynb

- [ ] I updated the docs.
PR to docs will certainly come before the next release, if not within a few days

This pull request adds a new feature to libra. @Palashio, could you please take a look at it?

### CODE REVIEW

**Comment by anas-awadalla** [comment]
@pranavnt This looks alright but I have a couple of concerns here that need to be addressed before merging. 1st cdqa is built on top of Pytorch which we decided to not be a part of our Libra for now. Also, you don't seem to be using our common framework of allowing the user to specify column names or extracting those from the command which would be a helpful feature to have in this setting. Also, there seems to be no modularity in the training loop which is inconsistent with other queries. Let's chat more on slack!

**Comment by pranavnt** [comment]
Sounds good! I'll close this PR and reopen once I fix those issues
